src/ApiManagement/ApiManagement.ServiceManagement/Commands/GetAzureApiManagementGatewayKey.cs

```diff
@@ -0,0 +1,60 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet("Get", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApiManagementGatewayKey", DefaultParameterSetName = GetByGatewayId, SupportsShouldProcess = true)]
+    [OutputType(typeof(PsApiManagementGatewayKey), ParameterSetName = new[] { GetByGatewayId })]
```

Comment:
this call is using http post but doesn't seem to make any changes on azure resources,
Can you please confirm?

If that's the case, please remove SupportShouldProcess

Reply:
Yes, no any changes on the resource.  SupportShouldProcess removed.